fuel/app/classes/materia/api/v1.php

```diff
@@ -68,6 +68,46 @@ static public function widget_instance_delete($inst_id)
 		return $inst->db_remove();
 	}
 
+	/**
+	 * @return object, contains properties indicating whether the current
+	 * user can edit the widget and a message object describing why, if not
+	 */
+	static public function widget_instance_edit_perms_verify(string $inst_id): \stdClass
+	{
+		$response = new \stdClass();
+		$response->msg = null;
+		$response->is_locked = true;
+		$response->can_publish = false;
+
+		if ( ! Util_Validator::is_valid_hash($inst_id)) $response->msg = Msg::invalid_input($inst_id);
+		else if (\Service_User::verify_session() !== true) $response->msg = Msg::no_login();
+		else if ( ! static::has_perms_to_inst($inst_id, [Perm::FULL])) $response->msg = Msg::no_perm();
+		else if ( ! ($inst = Widget_Instance_Manager::get($inst_id))) $response->msg = new Msg(Msg::ERROR, 'Widget instance does not exist.');
+
+		//msg property only set if something went wrong
+		if ( ! $response->msg)
+		{
+			$response->is_locked = ! Widget_Instance_Manager::locked_by_current_user($inst_id);
```

Comment:
The use case here seems to suggest we're checking if the instance is locked _for_ the current user, not _by_ the current user.

Reply:
Looks ok to me now